discord/utils/private.py

```diff
@@ -0,0 +1,549 @@
+        def __init__(self, data: Iterable[int], *, is_sorted: bool = False): ...
+
+    def __new__(cls, data: Iterable[int], *, is_sorted: bool = False):
+        return array.array.__new__(cls, "Q", data if is_sorted else sorted(data))  # type: ignore
+
+    def add(self, element: int) -> None:
+        i = bisect_left(self, element)
+        self.insert(i, element)
+
+    def get(self, element: int) -> int | None:
+        i = bisect_left(self, element)
+        return self[i] if i != len(self) and self[i] == element else None
+
+    def has(self, element: int) -> bool:
+        i = bisect_left(self, element)
+        return i != len(self) and self[i] == element
+
+
+def copy_doc(original: Callable) -> Callable[[T], T]:
+    def decorator(overridden: T) -> T:
+        overridden.__doc__ = original.__doc__
+        overridden.__signature__ = signature(original)  # type: ignore
+        return overridden
+
+    return decorator
+
+
+class SequenceProxy(collections.abc.Sequence, Generic[T_co]):
+    """Read-only proxy of a Sequence."""
+
+    def __init__(self, proxied: Sequence[T_co]):
+        self.__proxied = proxied
+
+    def __getitem__(self, idx: int) -> T_co:
+        return self.__proxied[idx]
+
+    def __len__(self) -> int:
+        return len(self.__proxied)
+
+    def __contains__(self, item: Any) -> bool:
+        return item in self.__proxied
+
+    def __iter__(self) -> Iterator[T_co]:
+        return iter(self.__proxied)
+
+    def __reversed__(self) -> Iterator[T_co]:
+        return reversed(self.__proxied)
+
+    def index(self, value: Any, *args, **kwargs) -> int:
+        return self.__proxied.index(value, *args, **kwargs)
+
+    def count(self, value: Any) -> int:
+        return self.__proxied.count(value)
+
+
+class CachedSlotProperty(Generic[T, T_co]):
+    def __init__(self, name: str, function: Callable[[T], T_co]) -> None:
+        self.name = name
+        self.function = function
+        self.__doc__ = getattr(function, "__doc__")
+
+    @overload
+    def __get__(self, instance: None, owner: type[T]) -> CachedSlotProperty[T, T_co]: ...
+
+    @overload
+    def __get__(self, instance: T, owner: type[T]) -> T_co: ...
+
+    def __get__(self, instance: T | None, owner: type[T]) -> Any:
+        if instance is None:
+            return self
+
+        try:
+            return getattr(instance, self.name)
+        except AttributeError:
+            value = self.function(instance)
+            setattr(instance, self.name, value)
+            return value
+
+
+def get_slots(cls: type[Any]) -> Iterator[str]:
+    for mro in reversed(cls.__mro__):
+        try:
+            yield from mro.__slots__
+        except AttributeError:
+            continue
+
+
+def cached_slot_property(
+    name: str,
+) -> Callable[[Callable[[T], T_co]], CachedSlotProperty[T, T_co]]:
+    def decorator(func: Callable[[T], T_co]) -> CachedSlotProperty[T, T_co]:
+        return CachedSlotProperty(name, func)
+
+    return decorator
+
+
+try:
```

Comment:
`_to_json` -> `to_json`
`_from_json` -> `from_json`
You have to update it in a few different imports as well